packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -685,6 +847,218 @@ contract BridgeRouter is
+
+        if (r.failedAt == 0) revert InvalidParams();
+
+        // State reads should not be retryable as they are read-only operations
+        if (r.operationType == BridgeTypes.OperationType.READ_STATE) {
+            revert UnsupportedOperationType();
+        }
+
+        // Use the original adapter - no override needed
+        address effectiveAdapter = r.adapter;
+
+        // Decode the original payload
+        bytes memory effectivePayload = r.operationPayload;
+
+        // Apply recipient override if provided
+        if (newRecipient != address(0)) {
+            // Apply recipient override to the payload
+            effectivePayload = _applyRecipientOverride(
+                r.operationType,
+                effectivePayload,
+                newRecipient
+            );
+        }
+
+        _validateRetryPayload(
+            r.operationType,
+            effectivePayload,
+            r.sourceChainId,
+            effectiveAdapter
+        );
+
+        try
+            this._processDelivery(
+                r.operationType,
+                effectivePayload,
+                effectiveAdapter
+            )
+        {
+            _clearFailedDelivery(operationId);
+            emit OperationRetrySucceeded(
+                operationId,
+                r.operationType,
+                effectiveAdapter
+            );
+        } catch (bytes memory err) {
+            // Do not create a new failure record; update existing metadata only
+            FailedDeliveryRecord storage existing = failedDeliveries[
+                operationId
+            ];
+            existing.failedAt = block.timestamp;
+
+            emit OperationRetryFailed(
+                operationId,
+                r.operationType,
+                effectiveAdapter,
+                err
+            );
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        RETRY RECIPIENT OVERRIDE FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Applies recipient override to the operation payload
+     * @param operationType Type of operation being retried
+     * @param originalPayload The original payload
+     * @param newRecipient The new recipient address
+     * @return modifiedPayload The payload with recipient override applied
+     * @dev This function only modifies the payload. Validations are performed separately by _validateRetryPayload
+     */
+    function _applyRecipientOverride(
+        BridgeTypes.OperationType operationType,
+        bytes memory originalPayload,
+        address newRecipient
+    ) internal pure returns (bytes memory modifiedPayload) {
+        if (operationType == BridgeTypes.OperationType.TRANSFER_ASSET) {
+            BridgeTypes.RelayedTransferParams memory params = abi.decode(
+                originalPayload,
+                (BridgeTypes.RelayedTransferParams)
+            );
+
```

Comment:
you can use `_validateArkFleetRelationship` here and get rid of bulk amount of code -> `_validateRetryPayload` , `_validateTransferPayload`, `_validateMessagePayload`